### ISSUE TITLE
Possibility to make `as const` unnecessary

### ISSUE THREAD

**Issue by fratzinger** [ThomasAribart/json-schema-to-ts#90]
Maybe there is a way to remove the necessity of `as const`. Please view the following example:

```ts
type Narrow<T> =
  | T extends Function ? T : never
  | T extends string | number | boolean | bigint ? T : never 
  | T extends [] ? [] : never 
  | { [K in keyof T]: Narrow<T[K]> }

declare function test<T>(a: Narrow<T>): T


const asConst = test(['string', 'literal', 123])
// of type ['string', 'literal', 123]
```
Play with it in [TS Playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAcghgJwQewO4B4AqA+KBeAKCigB8pMoIAPYCAOwBMBnKAMQFc6BjYAS2TpQA-OSgAuKHQgA3CAiKlR1WoxZNgCXnQDmiuuwC2AIzmKjyZABsIcQWSO9tW4MNESpshFAVkKy+sxQANoAuq6h4pIypj5QAN7BANJQWlAA1hAgyABm5CES8EhoWEGJIbgAvgQEDBBclojQ2Zw8-IK06ljYABRwBYgoGDgAlBKY1QRcAupQcEwAwtMueFAdwN1BAOTqmjqbADRQm5a8tAhwlgdQAIwATADMIcPVQA)

Source: https://twitter.com/hd_nvim/status/1578567206190780417

**Comment by ThomasAribart** [comment]
Hi @fratzinger and thanks for the issue !

I like the idea of staying only at type level and not impacting compiled code, which would happen using such a function. Also, this will not work with actual `.json` schemas once [`import schema from './schema.json' as const` will be implemented in TS](https://github.com/microsoft/TypeScript/issues/32063).

However, I do not see any harm in exposing such a function for those who want to use it, so yes, I'll implement and document it !

**Comment by ThomasAribart** [comment]
@fratzinger Should be available in 2.6.0 !

There is an issue with my JS exports though, which use `import`/`export` syntax that is not compatible with `cjs`. The issue has been raised in https://github.com/ThomasAribart/json-schema-to-ts/issues/97 so we can close this one if it's good for you !

**Comment by fratzinger** [comment]
Nice. Thanks for your work and this great package! 🙏
